tools/build_cros_gcc

```diff
@@ -1,178 +1,210 @@
+     echo \ Valid targets: 'aarch64-cros-linux-gnu', 'x86_64-cros-linux-gnu', 'i686-cros-linux-gnu', 'armv7l-cros-linux-gnueabihf'
+     exit 0;;
+   x-vx|x--versionx) echo build_cros_gcc version 1.27
+     exit 0;;
+   x-cx|x--cleanx) rm -rf build-binutils build-gcc build-glibc
+     rm -rf \
+     ./${GCC} \
+     ./${CLOOG} \
+     ./${GMP} \
+     ./${GLIBC} \
+     ./${BINUTILS} \
+     ./${MPFR} \
+     ./${MPC} \
+     ./linux-${KERNEL_VERSION} \
+     ./${ISL} \
+     ./${GLIBC}.tar.xz \
+     ./${GMP}.tar.xz \
+     ./${GCC}.tar.xz \
+     ./${BINUTILS}.tar.xz \
+     ./linux-${KERNEL_VERSION}.tar.xz \
+     ./${MPFR}.tar.xz \
+     ./${CLOOG}.tar.gz \
+     ./${ISL}.tar.bz2 \
+     ./${MPC}.tar.gz || true
+     exit 0;;
+   x-tx|x--constx)
+     echo KERNEL_VERSION=${KERNEL_VERSION}
+     echo MAJOR_VERSION=${MAJOR_VERSION}
+     echo CLOOG=${CLOOG}
+     echo ISL=${ISL}
+     echo MPC=${MPC}
+     echo GMP=${GMP}
+     echo MPFR=${MPFR}
+     echo GCC=${GCC}
+     echo BINUTILS=${BINUTILS}
+     echo GLIBC=${GLIBC}
+     exit 0;;
+   xarmv7l-cros-linux-gnueabihfx)
+     true;;
+   xaarch64-cros-linux-gnux)
+     true;;
+   xi686-cros-linux-gnux)
+     true;;
+   xx86_64-cros-linux-gnux)
+     true;;
+   xx)
+     $0 -h
+     exit 1;;
+   x*x)
+     echo -e "\033[1;31mInvalid target or option '$1'\033[0m"
+     $0 -h
+   exit 1;;
  esac
 
- # '\033[1;31' outputs red text. '\033[0m' reverts it to normal.
- command -v crew > /dev/null || { echo "\033[1;31mcrew command not found.\033[0m"; exit 1; }
+ export TARGET="$1" && shift
 
- if test ! $2; then
-   echo "$1" | grep x86_64 1>/dev/null
-   if [ $? == 0 ]; then
-   $0 $1 x86 $3;
-   exit 0;
+ if ! [[ "$1" =~ "^[0-9]+$" ]] &>/dev/null; then
+   export JOBS="$1"
+   shift
+ else
+   shift
+ fi
+ if [[ -d $1 ]] &>/dev/null; then
+   export PREFIX="$1"
+   shift
+ else
+   if ! [[ "$1" =~ "^[0-9]+$" ]] &>/dev/null; then
+     export JOBS="$1"
+     shift
+   else if { mkdir -p "$1" &>/dev/null; }; then
+     export PREFIX="$1"
+     shift
    fi
-   echo "$1" | grep i686 1>/dev/null
-   if [ $? == 0 ]; then
-   $0 $1 x86 $3;
-   exit 0;
+ fi
+ if [[ -z "${ARCH}" ]]; then
+   if { echo "${TARGET}" | grep x86_64 1>/dev/null; }; then
+     export ARCH="x86_64";
    fi
-   echo "$1" | grep armv7l 1>/dev/null
-   if [ $? == 0 ]; then
-   $0 $1 arm $3;
-   exit 0;
+   if { echo "${TARGET}" | grep i686 1>/dev/null; }; then
+     export ARCH="x86";
    fi
-   echo "$1" | grep aarch64 1>/dev/null
-   if [ $? == 0 ]; then
-   $0 $1 arm64 $3;
-   exit 0;
+   if { echo "${TARGET}" | grep armv7l 1>/dev/null; }; then
+     export ARCH="arm";
    fi
+   if { echo "${TARGET}" | grep aarch64 1>/dev/null; }; then
+     export ARCH="arm64";
+   fi
+ fi
+ if [[ -z "${PREFIX}" ]]; then
+   export PREFIX="$(crew const CREW_PREFIX | cut -d'=' -f2)"
  fi
- if test ! $3; then
-   $0 $1 $2 $(crew const CREW_PREFIX | cut -d'=' -f2)
-   exit 0
+ if [[ -z "${JOBS}" ]]; then
+   export JOBS="1"
```

Comment:
The script fails if you use more than one processor, just like you said.

Reply:
If the Chromebook has only 1 processor, `nproc --all` will return 1.  There is never a time `nproc --all` will report more processors than physically possible but it will return the most processors available which would be the most efficient default.  Actually, you don't need the export either and `JOBS` is not the true meaning of the variable.   `NPROC` might make more sense.

Reply:
JOBS means the number of jobs. What I mean is, building some of these in parallel make the compilation fail, or skip essential parts. I made it default to the safest option, only 1.

Reply:
Done.